pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -118,28 +122,28 @@ func (c *PruneController) excludedRevisionHistory(ctx context.Context, recorder
 				inProgressRevisions = append(inProgressRevisions, revisionNumber)
 
 			default:
-				// protect things you don't understand
 				unknownStatusRevisions = append(unknownStatusRevisions, revisionNumber)
 				recorder.Event("UnknownRevisionStatus", fmt.Sprintf("unknown status for revision %d: %v", revisionNumber, configMap.Data["status"]))
 			}
 		}
 	}
 
 	// Return early if nothing to prune
-	if len(succeededRevisions)+len(failedRevisions) == 0 {
+	if len(succeededRevisions)+len(failedRevisions)+len(unknownStatusRevisions) == 0 {
```

Comment:
If I understand the code correctly, the revision controller will never set  the configmap status out of InProgress today even if it is no longer being worked.

You probably want to set it to something like "NeverCompleted", right?

You should probably update that controller to set proper status on old revisions I'd think.

Sample data here:

```
- apiVersion: v1
  data:
    revision: "1"
    status: InProgress
  kind: ConfigMap
  metadata:
    creationTimestamp: "2019-12-05T18:24:01Z"
    name: revision-status-1
    namespace: openshift-kube-apiserver
    resourceVersion: "2760"
    selfLink: /api/v1/namespaces/openshift-kube-apiserver/configmaps/revision-status-1
    uid: 68df822f-178c-11ea-a4b6-02ba0195095b
- apiVersion: v1
  data:
    revision: "115"
    status: InProgress
  kind: ConfigMap
  metadata:
    creationTimestamp: "2020-04-22T18:41:10Z"
    name: revision-status-115
    namespace: openshift-kube-apiserver
    resourceVersion: "109420856"
    selfLink: /api/v1/namespaces/openshift-kube-apiserver/configmaps/revision-status-115
    uid: d3d6e579-130c-4e34-ac37-deb922fb3340
- apiVersion: v1
  data:
    revision: "121"
    status: InProgress
  kind: ConfigMap
  metadata:
    creationTimestamp: "2020-04-23T00:52:07Z"
    name: revision-status-121
    namespace: openshift-kube-apiserver
    resourceVersion: "109631969"
    selfLink: /api/v1/namespaces/openshift-kube-apiserver/configmaps/revision-status-121
    uid: 90bc20c5-1a8f-4a79-a9aa-605c440ab387
```

Side note, storing a revisionTriggered reason would be pretty nice.